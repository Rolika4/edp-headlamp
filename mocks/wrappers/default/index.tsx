/**
 * @jest-environment jsdom
 */

import { SnackbarProvider } from 'notistack';
import React from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';

export const TestWrapper: React.FC = ({ children }) => {
    const queryClient = new QueryClient();

    return (
        <QueryClientProvider client={queryClient}>
            <SnackbarProvider>{children}</SnackbarProvider>
        </QueryClientProvider>
    );
};
