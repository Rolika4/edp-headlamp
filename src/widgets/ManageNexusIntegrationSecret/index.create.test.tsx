/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../mocks/wrappers/default';
import { ManageNexusIntegrationSecret } from './index';

test('renders ManageNexusIntegrationSecret Create component', () => {
    render(
        <TestWrapper>
            <ManageNexusIntegrationSecret
                formData={{
                    currentElement: 'placeholder',
                    handleDeleteRow: jest.fn(),
                }}
            />
        </TestWrapper>
    );

    const dialog = screen.getByTestId('form');
    expect(dialog).toMatchSnapshot();
});
