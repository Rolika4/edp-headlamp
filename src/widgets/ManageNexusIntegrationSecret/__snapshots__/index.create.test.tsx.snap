// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ManageNexusIntegrationSecret Create component 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  data-testid="form"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            style="display: flex;"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <span
                  class="makeStyles-labelWrap-1"
                >
                  <span
                    class="MuiTypography-root makeStyles-label-2 MuiTypography-body1"
                  >
                    URL
                  </span>
                  <div
                    class="MuiBox-root MuiBox-root-3"
                    style="height: 1.25rem;"
                  />
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                style="display: flex; align-items: flex-end;"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        name="url"
                        placeholder="Enter URL"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            style="display: flex;"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <span
                  class="makeStyles-labelWrap-1"
                >
                  <span
                    class="MuiTypography-root makeStyles-label-2 MuiTypography-body1"
                  >
                    User
                  </span>
                  <div
                    class="MuiBox-root MuiBox-root-4"
                    style="height: 1.25rem;"
                  />
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                style="display: flex; align-items: flex-end;"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        name="username"
                        placeholder="Enter user name"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            style="display: flex;"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <span
                  class="makeStyles-labelWrap-1"
                >
                  <span
                    class="MuiTypography-root makeStyles-label-2 MuiTypography-body1"
                  >
                    Password
                  </span>
                  <div
                    class="MuiBox-root MuiBox-root-5"
                    style="height: 1.25rem;"
                  />
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                style="display: flex; align-items: flex-end;"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                        name="password"
                        placeholder="Enter password"
                        type="password"
                        value=""
                      />
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <span />
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-content-xs-space-between"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            cancel
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                undo changes
              </span>
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
